dhall/src/Dhall/Pretty/Internal.hs

```diff
@@ -1277,6 +1289,22 @@ escapeTrailingSingleQuote chunks@(Chunks as b) =
 pretty_ :: Pretty a => a -> Text
 pretty_ = prettyToStrictText
 
+{- This utility function converts
+   `{ x = { y = { z = 1 } } }` to `{ x.y.z. = 1 }`
+-}
+consolidateRecordLiteral :: Map Text (Expr s a) -> Map [Text] (Expr s a)
```

Comment:
Looks like we could use `NonEmpty` here – if only to avoid that the next reader wonders why it doesn't! ;)

```suggestion
consolidateRecordLiteral :: Map Text (Expr s a) -> Map (NonEmpty Text) (Expr s a)
```